R/criterion.R

```diff
@@ -145,7 +137,22 @@ print.root_criterion <- function(x, ...) {
   stopifnot(is.root_criterion(y))
 
   root_criterion(
-    c(x$testfun, y$testfun),
-    c(x$desc, y$desc)
+    testfun = function(path) { x$testfun(path) | y$testfun(path) },
```

Comment:
I think this is a bit more extensible, but changes the expected inputs for others who were creating their own `root_criterion`. I don't think that's a huge concern, though, as I don't think the old system could have been extended beyond what was already there (vanilla function and `|`)

Reply:
As far as I know, there's no `||`. I always try it (since I'm used to C++ & JavaScript), and it always fails. Does it exist? I don't see it in the few pages I searched for "R boolean operators" (e.g. http://www.statmethods.net/management/operators.html)

Reply:
`||` only works for atomic booleans, whereas `|` also works for vectors. Try `TRUE || FALSE` or `?"||"`.

The main question here is if `testfun()` is vectorized or not. As far as I remember it's not.

Reply:
It doesn't look to be vectorized. I'll switch to `||` and `&&`.